### PR TITLE
Radio ui conversion to tgui

### PR DESCRIPTION
**Pull request by Luxizzle**
## About the PR 

Converts the UI's on radios to tgui windows.

dreamseeker_L2IxVsfZPr

And with wire panel open

dreamseeker_YykjTlVYKp

## Why's this needed? 

More tgui makes me happy.

## Changelog

```changelog
(u)Luxizzle
(*)Radio & Intercom UI's have been converted to tgui.
```

### CODE REVIEW

File: code/obj/item/device/radio.dm [hunk 1]
```diff
@@ -108,67 +108,61 @@ var/list/headset_channel_lookup
...
+		"lockedFrequency" = src.locked_frequency,
+		"secureFrequencies" = frequencies,
+		"wires" = src.wires,
+		"modifiable" = src.b_stat,
+	)
+
+/obj/item/device/radio/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if (.)
+		return .
```
Comment (ZeWaka) [inline code/obj/item/device/radio.dm:152]:
```suggestion
		return
```
not needed, `.` is the default return value - will lead to cargo culting

File: code/obj/item/device/radio.dm [hunk 2]
```diff
@@ -108,67 +108,61 @@ var/list/headset_channel_lookup
...
+
+	switch(action)
+		if ("set-frequency")
+			if (src.locked_frequency)
+				return FALSE
+			set_frequency(sanitize_frequency(params["value"]))
+			// TODO: traitor uplinks
```
Comment (ZeWaka) [inline code/obj/item/device/radio.dm:159]: why todo? looks handled below

File: tgui/packages/tgui/interfaces/Radio/index.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  Knob,
+  LabeledList,
+  Section,
+  Stack,
+  Table,
+} from "../../components";
+import { Window } from "../../layouts";
+import { RadioData, RadioWires } from "./type";
+
+const formatFrequency = f => {
```
Comment (ZeWaka) [inline tgui/packages/tgui/interfaces/Radio/index.tsx:15]: would rather this go in the format file/module/whatever

File: tgui/packages/tgui/interfaces/Radio/index.tsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  f = Math.round(f);
+  return Math.floor(f / 10) + "." + (f % 10);
+};
+
+export const Radio = (_props, context) => {
+  const { data, act } = useBackend<RadioData>(context);
+
+  const setFrequency = (value: number, finish: boolean) => {
```
Comment (ZeWaka) [inline tgui/packages/tgui/interfaces/Radio/index.tsx:23]: what is `finish`

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/obj/item/device/radio.dm (hunk 2 above)
Comment (Luxizzle) [reply code/obj/item/device/radio.dm:159]: woops

File: tgui/packages/tgui/interfaces/Radio/index.tsx (hunk 4 above)
Comment (Luxizzle) [reply tgui/packages/tgui/interfaces/Radio/index.tsx:23]: finish is set to true on the onChange handler. Handles traitor uplinks on the byond side so you cant skim through all the frequencies.

**Review by ZeWaka — APPROVED** [review]

File: tgui/packages/tgui/interfaces/Radio/index.tsx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+                </LabeledList.Item>
+                <LabeledList.Item label="Frequency">
+                  <Stack align="center">
+                    <Stack.Item>
+                      {!data.lockedFrequency && (
+                        <Knob
+                          animated
+                          value={data.frequency}
+                          minValue={1441}
+                          maxValue={1489}
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/interfaces/Radio/index.tsx:54]: Throw these as constants at the top of the file, `MIN_FREQUENCY` and `MAX_FREQUENCY` or something.

File: tgui/packages/tgui/interfaces/Radio/index.tsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+                  </LabeledList.Item>
+                  <LabeledList.Item label="Blue Wire" labelColor="blue">
+                    <Button
+                      color="blue"
+                      onClick={() =>
+                        act("toggle-wire", { wire: RadioWires.Signal })}
+                    >
+                      {data.wires & RadioWires.Signal ? "Cut" : "Mend"}
+                    </Button>
+                  </LabeledList.Item>
```
Comment (mordent-goonstation) [inline tgui/packages/tgui/interfaces/Radio/index.tsx:126]: I feel like making the wires part of the data at some point would be good. I understand this is not needed with the current implementation.

**Review by mordent-goonstation — CHANGES_REQUESTED** [review]
Looks good overall, just a couple minor bits and good to merge.

**Review by mordent-goonstation — APPROVED** [review]

**Comment by ZeWaka** [comment]
haha get rekt mordent

**Pull request merged**